File: frontend/src/api/publish.ts
import { api } from './index'

export interface Publication {
  id: number
  content_id: number
  platform: string
  publish_type: 'image_post' | 'video_post'
  status: 'draft' | 'queued' | 'running' | 'scheduled' | 'completed' | 'failed'
  title?: string
  body: string
  scheduled_at?: string
  published_at?: string
  external_post_id?: string
  error_message?: string
  request_payload?: Record<string, unknown> | null
  response_payload?: Record<string, unknown> | null
  created_at?: string
  updated_at?: string
}

export interface PublishPayload {
  content_id: number
  publish_type: 'image_post' | 'video_post'
  title?: string
  content?: string
  media_ids?: number[]
  scheduled_at?: string
  tags?: string[]
  visibility?: 'public' | 'self-only' | 'friends-only'
  is_original?: boolean
}

export interface PublishActionResponse {
  publication: Publication
  job_id: string
}

export interface XiaohongshuLoginStatus {
  connected: boolean
  status_text: string
  details?: Record<string, unknown> | null
}

export async function getXiaohongshuLoginStatus() {
  const { data } = await api.get<XiaohongshuLoginStatus>('/publish/xiaohongshu/login-status')
  return data
}

export async function publishToXiaohongshu(payload: PublishPayload) {
  const { data } = await api.post<PublishActionResponse>('/publish/xiaohongshu', payload)
  return data
}

export async function scheduleXiaohongshuPublication(payload: PublishPayload) {
  const { data } = await api.post<PublishActionResponse>('/publish/xiaohongshu/schedule', payload)
  return data
}

export async function getContentPublications(contentId: number) {
  const { data } = await api.get<Publication[]>(`/publish/content/${contentId}`)
  return data
}

File: src/api/schemas/publish.py
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field

PublicationStatus = Literal["draft", "queued", "running", "scheduled", "completed", "failed"]
PublishType = Literal["image_post", "video_post"]


class XiaohongshuPublishRequest(BaseModel):
    content_id: int = Field(..., gt=0)
    publish_type: PublishType
    title: str | None = None
    content: str | None = None
    media_ids: list[int] | None = None
    scheduled_at: datetime | None = None
    tags: list[str] | None = None
    visibility: Literal["public", "self-only", "friends-only"] = "public"
    is_original: bool = False


class PublicationResponse(BaseModel):
    id: int
    content_id: int
    platform: str
    publish_type: PublishType
    status: PublicationStatus
    title: str | None = None
    body: str
    scheduled_at: str | None = None
    published_at: str | None = None
    external_post_id: str | None = None
    error_message: str | None = None
    request_payload: dict[str, Any] | None = None
    response_payload: dict[str, Any] | None = None
    created_at: str | None = None
    updated_at: str | None = None


class PublishActionResponse(BaseModel):
    publication: PublicationResponse
    job_id: str


class XiaohongshuLoginStatusResponse(BaseModel):
    connected: bool
    status_text: str = ""
    details: dict[str, Any] | None = None
